services/network/public/cpp/system_request_handler.h

```diff
@@ -0,0 +1,52 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_SERVICES_NETWORK_PUBLIC_CPP_SYSTEM_REQUEST_HANDLER_H_
+#define BRAVE_SERVICES_NETWORK_PUBLIC_CPP_SYSTEM_REQUEST_HANDLER_H_
+
+#include "base/callback.h"
+#include "base/component_export.h"
+
+namespace network {
+struct ResourceRequest;
+}  // namespace network
+
+namespace base {
```

Comment:
`base` is one place where you can just add includes directly and not worry about forward declarations